app/uk/gov/hmrc/payetaxcalculatorfrontend/quickmodel/TaxResult.scala

```diff
@@ -79,6 +79,20 @@ object TaxResult {
       case _ => -1
     }
 
+  def incomeTax(maxTaxAmount: BigDecimal, taxCategory: Seq[TaxCategory]): BigDecimal = {
```

Comment:
It isn't already a clean model. The purpose of this method is only to pull the complex Scala evaluations out of the templete. The less parameters the better is in that perspective.